src/composables/use-hide-on-click-outside.js

```diff
@@ -37,9 +37,7 @@ function useMouseDownTargetRef({
       const document = getDocument(popover)
       const onMouseDown = (event) => (mouseDownTargetRef.value = event.target)
       document.addEventListener('mousedown', onMouseDown)
-      onInvalidate(() => {
-        document.addEventListener('mousedown', onMouseDown)
-      })
+      onInvalidate(() => {})
```

Comment:
```suggestion
      onInvalidate(() => {
        document.removeEventListener('mousedown', onMouseDown)
      })
```
We need to actually add `removeEventListener` here so that we don't have any memory leaks. You can read more about it here: https://www.ditdot.hr/en/causes-of-memory-leaks-in-javascript-and-how-to-avoid-them#event-listeners